net.minecraft.ReportedException: Exception generating new chunk Caused by: java.lang.NullPointerException: Cannot load from int array because "this.surfaceHeight" is null

**What happened**
I Had played HardRock 4 tfc, in winter period tornadoes by weather2 suddenly generated and had passed my chunks, after it passed it has tossed rocks and other stuff. After laterly server was restarted and i was unnable to join server with that exception below. My assumption that rocks that was loaded in aactive chunk, may left loaded then they was converted to entities, after restart they may have to leave their loaded state or so, but chunks there they did end up - not, so there is exception appears https://github.com/TerraFirmaCraft/TerraFirmaCraft/blob/97a9f10129273aa6bb00acd76eaf243642c20794/src/main/java/net/dries007/tfc/world/chunkdata/RockData.java#L57.

Crash report:
```
[09:13:44] [Server thread/INFO] [patchouli/]: Sending reload packet to clients
[09:13:44] [Server thread/WARN] [ModernFix/]: Dedicated server took 84.483 seconds to load
[09:13:46] [Worker-Main-6/INFO] [citadel/]: vanilla surface rule behavior unchanged
[09:13:48] [Server thread/ERROR] [minecraft/BlockableEventLoop]: Error executing task on Chunk source main thread executor for minecraft:overworld
minecraft-tfc  | net.minecraft.ReportedException: Exception generating new chunk
minecraft-tfc  |        at net.minecraft.server.level.ChunkMap.m_214912_(ChunkMap.java:658) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betterchunkloading.mixins.json:ChunkMapViewDistanceFixedTwo,pl:mixin:APP:modernfix-modernfix.mixins.json:bugfix.paper_chunk_patches.ChunkMapMixin,pl:mixin:APP:modernfix-modernfix.mixins.json:bugfix.chunk_deadlock.ChunkMapLoadMixin,pl:mixin:APP:chunkpregen.mixins.json:common.storage.ChunkManagerMixin,pl:mixin:APP:adastra-common.mixins.json:common.multipart.ChunkMapMixin,pl:mixin:APP:tfc.mixins.json:ChunkMapMixin,pl:mixin:APP:tfc.mixins.json:accessor.ChunkMapAccessor,pl:mixin:APP:railways.mixins.json:ChunkMapAccessor,pl:mixin:APP:ftbchunks-common.mixins.json:ChunkMapMixin,pl:mixin:APP:lychee.mixins.json:ChunkMapAccess,pl:mixin:APP:securitycraft.mixins.json:camera.ChunkMapMixin,pl:mixin:APP:railways-common.mixins.json:conductor_possession.ChunkMapMixin,pl:mixin:A}
minecraft-tfc  |        at net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:156) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
minecraft-tfc  |        at net.minecraft.server.level.ServerChunkCache$MainThreadExecutor.m_6367_(ServerChunkCache.java:541) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading,pl:accesstransformer:B}
minecraft-tfc  |        at net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:130) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
minecraft-tfc  |        at net.minecraft.server.level.ServerChunkCache$MainThreadExecutor.m_7245_(ServerChunkCache.java:549) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading,pl:accesstransformer:B}
minecraft-tfc  |        at net.minecraft.util.thread.BlockableEventLoop.m_18701_(BlockableEventLoop.java:139) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
minecraft-tfc  |        at net.minecraft.server.level.ServerChunkCache.m_7587_(ServerChunkCache.java:131) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:cupboard.mixins.json:ChunkLoadDebug,pl:mixin:APP:betterchunkloading.mixins.json:ServerChunkCacheMixin,pl:mixin:APP:chunkpregen.mixins.json:common.storage.ServerChunkProviderMixin,pl:mixin:APP:modernfix-modernfix.mixins.json:bugfix.chunk_deadlock.ServerChunkCache_CurrentLoadingMixin,pl:mixin:A}
minecraft-tfc  |        at net.minecraft.world.level.Level.m_6522_(Level.java:187) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:twilightforest:cloud,re:computing_frames,pl:accesstransformer:B,xf:fml:twilightforest:cloud,re:classloading,pl:accesstransformer:B,xf:fml:twilightforest:cloud,pl:mixin:APP:adastra.mixins.json:common.multipart.LevelMixin,pl:mixin:APP:tfc.mixins.json:LevelMixin,pl:mixin:APP:citadel.mixins.json:LevelMixin,pl:mixin:APP:kubejs-common.mixins.json:LevelMixin,pl:mixin:APP:observable.common.json:LevelMixin,pl:mixin:A}
minecraft-tfc  |        at net.minecraft.world.level.LevelReader.m_46819_(LevelReader.java:129) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:computing_frames,re:mixin,re:classloading}

minecraft-tfc  |        at net.minecraft.world.level.Level.m_6325_(Level.java:182) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:twilightforest:cloud,re:computing_frames,pl:accesstransformer:B,xf:fml:twilightforest:cloud,re:classloading,pl:accesstransformer:B,xf:fml:twilightforest:cloud,pl:mixin:APP:adastra.mixins.json:common.multipart.LevelMixin,pl:mixin:APP:tfc.mixins.json:LevelMixin,pl:mixin:APP:citadel.mixins.json:LevelMixin,pl:mixin:APP:kubejs-common.mixins.json:LevelMixin,pl:mixin:APP:observable.common.json:LevelMixin,pl:mixin:A}
minecraft-tfc  |        at net.minecraft.world.level.Level.m_8055_(Level.java:355) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:twilightforest:cloud,re:computing_frames,pl:accesstransformer:B,xf:fml:twilightforest:cloud,re:classloading,pl:accesstransformer:B,xf:fml:twilightforest:cloud,pl:mixin:APP:adastra.mixins.json:common.multipart.LevelMixin,pl:mixin:APP:tfc.mixins.json:LevelMixin,pl:mixin:APP:citadel.mixins.json:LevelMixin,pl:mixin:APP:kubejs-common.mixins.json:LevelMixin,pl:mixin:APP:observable.common.json:LevelMixin,pl:mixin:A}
minecraft-tfc  |        at weather2.weathersystem.storm.TornadoHelper.tick(TornadoHelper.java:337) ~[weather2-1.20.1-2.8.3.jar%23720!/:1.20.1-2.8.3] {re:mixin,pl:runtimedistcleaner:A,re:classloading,pl:mixin:APP:tfcweather.mixins.json:weather2.TornadoHelperMixin,pl:mixin:A,pl:runtimedistcleaner:A}
minecraft-tfc  |        at weather2.weathersystem.storm.StormObject.tick(StormObject.java:744) ~[weather2-1.20.1-2.8.3.jar%23720!/:1.20.1-2.8.3] {re:mixin,pl:runtimedistcleaner:A,re:classloading,pl:mixin:APP:tfcweather.mixins.json:weather2.StormObjectMixin,pl:mixin:A,pl:runtimedistcleaner:A}
minecraft-tfc  |        at weather2.weathersystem.WeatherManager.tick(WeatherManager.java:64) ~[weather2-1.20.1-2.8.3.jar%23720!/:1.20.1-2.8.3] {re:classloading,re:mixin}
minecraft-tfc  |        at weather2.weathersystem.WeatherManagerServer.tick(WeatherManagerServer.java:889) ~[weather2-1.20.1-2.8.3.jar%23720!/:1.20.1-2.8.3] {re:mixin,re:classloading,pl:mixin:APP:tfcweather.mixins.json:weather2.WeatherManagerServerMixin,pl:mixin:A}
minecraft-tfc  |        at weather2.ServerTickHandler.tickServer(ServerTickHandler.java:68) ~[weather2-1.20.1-2.8.3.jar%23720!/:1.20.1-2.8.3] {re:classloading,re:mixin}
minecraft-tfc  |        at weather2.__ServerTickHandler_tickServer_ServerTickEvent.invoke(.dynamic) ~[weather2-1.20.1-2.8.3.jar%23720!/:1.20.1-2.8.3] {re:classloading,pl:eventbus:B}
minecraft-tfc  |        at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:73) ~[eventbus-6.0.5.jar%2352!/:?] {}
minecraft-tfc  |        at net.minecraftforge.eventbus.EventBus.post(EventBus.java:315) ~[eventbus-6.0.5.jar%2352!/:?] {}
minecraft-tfc  |        at net.minecraftforge.eventbus.EventBus.post(EventBus.java:296) ~[eventbus-6.0.5.jar%2352!/:?] {}
minecraft-tfc  |        at net.minecraftforge.event.ForgeEventFactory.onPreServerTick(ForgeEventFactory.java:970) ~[forge-1.20.1-47.4.13-universal.jar%23736!/:?] {re:mixin,re:classloading,pl:mixin:APP:modernfix-modernfix.mixins.json:perf.potential_spawns_alloc.ForgeEventFactoryMixin,pl:mixin:A}
minecraft-tfc  |        at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:812) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
minecraft-tfc  |        at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:661) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}
minecraft-tfc  |        at net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:251) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:A}

minecraft-tfc  |        at java.lang.Thread.run(Thread.java:840) ~[?:?] {re:mixin}
minecraft-tfc  | Caused by: java.lang.NullPointerException: Cannot load from int array because "this.surfaceHeight" is null
minecraft-tfc  |        at net.dries007.tfc.world.chunkdata.RockData.getRock(RockData.java:57) ~[TerraFirmaCraft-Forge-1.20.1-3.2.21.jar%23659!/:3.2.21] {re:classloading}
minecraft-tfc  |        at net.dries007.tfc.world.chunkdata.RockData.getRock(RockData.java:51) ~[TerraFirmaCraft-Forge-1.20.1-3.2.21.jar%23659!/:3.2.21] {re:classloading}
minecraft-tfc  |        at net.dries007.tfc.world.feature.LooseRockFeature.m_142674_(LooseRockFeature.java:47) ~[TerraFirmaCraft-Forge-1.20.1-3.2.21.jar%23659!/:3.2.21] {re:classloading}
minecraft-tfc  |        at net.minecraft.world.level.levelgen.feature.Feature.m_225028_(Feature.java:155) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,re:classloading,pl:mixin:APP:farmersdelight.mixins.json:KeepRichSoilGiantTreeMixin,pl:mixin:A}
minecraft-tfc  |        at net.minecraft.world.level.levelgen.feature.ConfiguredFeature.m_224953_(ConfiguredFeature.java:28) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading}
minecraft-tfc  |        at net.minecraft.world.level.levelgen.placement.PlacedFeature.m_226362_(PlacedFeature.java:54) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading,pl:accesstransformer:B}
minecraft-tfc  |        at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:411) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:411) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596) ~[?:?] {}
minecraft-tfc  |        at net.minecraft.world.level.levelgen.placement.PlacedFeature.m_226368_(PlacedFeature.java:53) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading,pl:accesstransformer:B}
minecraft-tfc  |        at net.minecraft.world.level.levelgen.placement.PlacedFeature.m_226357_(PlacedFeature.java:38) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading,pl:accesstransformer:B}
minecraft-tfc  |        at net.minecraft.world.level.levelgen.feature.RandomPatchFeature.m_142674_(RandomPatchFeature.java:28) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading}
minecraft-tfc  |        at net.minecraft.world.level.levelgen.feature.Feature.m_225028_(Feature.java:155) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,re:classloading,pl:mixin:APP:farmersdelight.mixins.json:KeepRichSoilGiantTreeMixin,pl:mixin:A}
minecraft-tfc  |        at net.minecraft.world.level.levelgen.feature.ConfiguredFeature.m_224953_(ConfiguredFeature.java:28) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading}
minecraft-tfc  |        at net.minecraft.world.level.levelgen.placement.PlacedFeature.m_226362_(PlacedFeature.java:54) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading,pl:accesstransformer:B}
minecraft-tfc  |        at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183) ~[?:?] {}

minecraft-tfc  |        at java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:411) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:411) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?] {}
minecraft-tfc  |        at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596) ~[?:?] {}
minecraft-tfc  |        at net.minecraft.world.level.levelgen.placement.PlacedFeature.m_226368_(PlacedFeature.java:53) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading,pl:accesstransformer:B}
minecraft-tfc  |        at net.minecraft.world.level.levelgen.placement.PlacedFeature.m_226377_(PlacedFeature.java:42) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading,pl:accesstransformer:B}
minecraft-tfc  |        at net.dries007.tfc.world.TFCChunkGenerator.m_213609_(TFCChunkGenerator.java:369) ~[TerraFirmaCraft-Forge-1.20.1-3.2.21.jar%23659!/:3.2.21] {re:classloading}
minecraft-tfc  |        at net.minecraft.world.level.chunk.ChunkStatus.m_279978_(ChunkStatus.java:108) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
minecraft-tfc  |        at net.minecraft.world.level.chunk.ChunkStatus$SimpleGenerationTask.m_214024_(ChunkStatus.java:309) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading}
minecraft-tfc  |        at net.minecraft.world.level.chunk.ChunkStatus.m_280308_(ChunkStatus.java:252) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
minecraft-tfc  |        at net.minecraft.server.level.ChunkMap.lambda$scheduleChunkGeneration$27(ChunkMap.java:643) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betterchunkloading.mixins.json:ChunkMapViewDistanceFixedTwo,pl:mixin:APP:modernfix-modernfix.mixins.json:bugfix.paper_chunk_patches.ChunkMapMixin,pl:mixin:APP:modernfix-modernfix.mixins.json:bugfix.chunk_deadlock.ChunkMapLoadMixin,pl:mixin:APP:chunkpregen.mixins.json:common.storage.ChunkManagerMixin,pl:mixin:APP:adastra-common.mixins.json:common.multipart.ChunkMapMixin,pl:mixin:APP:tfc.mixins.json:ChunkMapMixin,pl:mixin:APP:tfc.mixins.json:accessor.ChunkMapAccessor,pl:mixin:APP:railways.mixins.json:ChunkMapAccessor,pl:mixin:APP:ftbchunks-common.mixins.json:ChunkMapMixin,pl:mixin:APP:lychee.mixins.json:ChunkMapAccess,pl:mixin:APP:securitycraft.mixins.json:camera.ChunkMapMixin,pl:mixin:APP:railways-common.mixins.json:conductor_possession.ChunkMapMixin,pl:mixin:A}
minecraft-tfc  |        at com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[datafixerupper-6.0.8.jar%2377!/:?] {}

minecraft-tfc  |        at net.minecraft.server.level.ChunkMap.lambda$scheduleChunkGeneration$29(ChunkMap.java:634) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betterchunkloading.mixins.json:ChunkMapViewDistanceFixedTwo,pl:mixin:APP:modernfix-modernfix.mixins.json:bugfix.paper_chunk_patches.ChunkMapMixin,pl:mixin:APP:modernfix-modernfix.mixins.json:bugfix.chunk_deadlock.ChunkMapLoadMixin,pl:mixin:APP:chunkpregen.mixins.json:common.storage.ChunkManagerMixin,pl:mixin:APP:adastra-common.mixins.json:common.multipart.ChunkMapMixin,pl:mixin:APP:tfc.mixins.json:ChunkMapMixin,pl:mixin:APP:tfc.mixins.json:accessor.ChunkMapAccessor,pl:mixin:APP:railways.mixins.json:ChunkMapAccessor,pl:mixin:APP:ftbchunks-common.mixins.json:ChunkMapMixin,pl:mixin:APP:lychee.mixins.json:ChunkMapAccess,pl:mixin:APP:securitycraft.mixins.json:camera.ChunkMapMixin,pl:mixin:APP:railways-common.mixins.json:conductor_possession.ChunkMapMixin,pl:mixin:A}
minecraft-tfc  |        at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150) ~[?:?] {}
minecraft-tfc  |        at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482) ~[?:?] {}
minecraft-tfc  |        at net.minecraft.server.level.ChunkTaskPriorityQueueSorter.m_143188_(ChunkTaskPriorityQueueSorter.java:62) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,re:classloading,pl:mixin:APP:betterchunkloading.mixins.json:ChunkTaskPriorityQueueSorterMixin,pl:mixin:A}
minecraft-tfc  |        at net.minecraft.util.thread.ProcessorMailbox.m_18759_(ProcessorMailbox.java:91) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading}
minecraft-tfc  |        at net.minecraft.util.thread.ProcessorMailbox.m_18747_(ProcessorMailbox.java:146) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading}
minecraft-tfc  |        at net.minecraft.util.thread.ProcessorMailbox.run(ProcessorMailbox.java:102) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading}
minecraft-tfc  |        at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395) ~[?:?] {}
minecraft-tfc  |        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[?:?] {}
minecraft-tfc  |        at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[?:?] {}
minecraft-tfc  |        at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[?:?] {re:mixin,re:computing_frames}
minecraft-tfc  |        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[?:?] {re:mixin,re:computing_frames}
minecraft-tfc  |        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[?:?] {re:mixin}
[09:13:48] [Server thread/ERROR] [ne.mi.ev.EventBus/EVENTBUS]: Exception caught during firing event: net.minecraft.ReportedException: Exception generating new chunk
minecraft-tfc  |        Index: 6
minecraft-tfc  |        Listeners:
minecraft-tfc  |                0: HIGHEST
minecraft-tfc  |                1: net.minecraftforge.eventbus.EventBus$$Lambda$1589/0x000071e4fd81c930@72b78f3d
minecraft-tfc  |                2: HIGH
minecraft-tfc  |                3: ASM: class dev.architectury.event.forge.EventHandlerImplCommon event(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                4: NORMAL
minecraft-tfc  |                5: ASM: class io.redspace.pvp_flagging.core.PlayerFlagManager handleServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                6: ASM: class weather2.ServerTickHandler tickServer(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                7: ASM: com.vecoo.extralib.task.TaskTimer$EventHandler@23917c67 onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                8: ASM: net.mcreator.butchersdelightfoods.ButchersdelightfoodsMod@69b33d81 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V

minecraft-tfc  |                9: ASM: pregenerator.common.tracker.ServerTracker@3f94ab31 onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                10: ASM: net.mcreator.butchersdelight.ButchersdelightMod@30f16ef8 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                11: ASM: class com.mrcrayfish.guns.common.DelayedTask onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                12: ASM: net.antrolgaming.hudtexts.AgsHudTextsMod@53bc7a57 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                13: ASM: me.desht.pneumaticcraft.common.amadron.AmadronEventListener@37918074 amadronHousekeeping(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                14: ASM: class me.desht.pneumaticcraft.common.block.entity.PressureChamberWallBlockEntity$PressureWallStateManager onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                15: net.minecraftforge.eventbus.EventBus$$Lambda$1589/0x000071e4fd81c930@3b8975c6
minecraft-tfc  |                16: ASM: class com.blamejared.crafttweaker.impl.event.CTCommonEventHandler serverTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                17: ASM: net.minecraftforge.common.ForgeInternalHandler@44872af1 onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                18: ASM: class net.geforcemods.securitycraft.SCEventHandler onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                19: ASM: class noobanidus.mods.lootr.entity.EntityTicker onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                20: ASM: class noobanidus.mods.lootr.block.entities.TileTicker serverTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                21: ASM: class com.momosoftworks.coldsweat.common.capability.handler.EntityTempManager$Events cleanRemovedEntities(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                22: ASM: net.mcreator.trashbinmod.TrashbinmodMod@1905296b tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                23: ASM: carbonconfiglib.impl.internal.EventHandler@1335aeac onServerTickEvent(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                24: ASM: dttfc_seeds.DttfcSeedsMod@3c1760b5 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                25: net.minecraftforge.eventbus.EventBus$$Lambda$1589/0x000071e4fd81c930@5c0d8c27
minecraft-tfc  |                26: ASM: tfclumberjack.TfcLumberjackMod@1196e811 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                27: ASM: tfcdebark.TfcDebarkMod@673939cc tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                28: net.minecraftforge.eventbus.EventBus$$Lambda$1589/0x000071e4fd81c930@6a1a5218
minecraft-tfc  |                29: ASM: class com.connectivity.event.EventHandler onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                30: ASM: insane96mcp.insanelib.module.base.TimeStopNoPlayerOnline@5edb29b4 serverTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                31: ASM: class insane96mcp.insanelib.world.scheduled.ScheduledTasks onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                32: ASM: tfcstonetools.TfcStoneToolsMod@14af39e9 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                33: ASM: create_jetpack_curios.CreateJetpackCuriosMod@2c3e0d6f tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                34: ASM: farmersdelight_tfc.FarmersdelightTfcMod@38389f1c tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V

minecraft-tfc  |                35: ASM: htm.HtmMod@1f2fd1a0 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                36: ASM: tfcoreexcavation.TfcOreExcavationMod@5e35f677 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                37: ASM: justblahaj.JustBlahajMod@47f4a369 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                38: ASM: animaltrap.AnimaltrapMod@3a39e336 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                39: ASM: animaltraptfc.AnimaltraptfcMod@626392b6 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                40: ASM: spartanweaponrytfc.SpartanWeaponryTfcMod@a1737a6 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                41: ASM: class com.cupboard.event.EventHandler onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                42: ASM: net.mcreator.supersifting.SuperSiftingMod@5c205436 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                43: ASM: day_counter.DayCounterMod@6df2c956 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                44: ASM: com.mrcrayfish.framework.event.ForgeEvents@609693cb onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                45: ASM: com.kleiders.kleidersplayerrenderer.KleidersCustomRendererMod@17ba0e8e tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                46: ASM: net.mcreator.enlightened_end.EnlightenedEndMod@2976a670 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                47: ASM: class com.simibubi.create.foundation.events.CommonEvents onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                48: ASM: create_hc.CreateHcMod@38923d1d tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                49: ASM: net.donne431.create_curios_backtank.CreateCuriosBacktankMod@237e1abe tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                50: ASM: coralstfc.CoralstfcMod@76428b1c tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                51: ASM: vinery_tfc.VineryTfcMod@32077ff7 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                52: ASM: hardrock_samples.HardrockSamplesMod@4e3cfbcf tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                53: ASM: tfcloot.TfcLootMod@5f811dc7 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                54: ASM: tfcstronghold.TfcStrongholdMod@1da9b451 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                55: ASM: tfcjungletemple.TfcJungleTempleMod@637dcd08 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                56: ASM: tfc_dungeons_arise.TfcDungeonsAriseMod@70b95bdc tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                57: ASM: tfcswamphut.TfcSwampHutMod@5db5b087 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                58: ASM: tfctrees.TfcTreesMod@664625aa tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                59: ASM: tfcdeserttemple.TfcDesertTempleMod@3de0a214 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                60: ASM: spartanshieldstfc.SpartanShieldsTfcMod@1172d474 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                61: ASM: net.mcreator.createstuffadditions.CreateSaMod@70b35da6 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                62: ASM: class sonar.fluxnetworks.register.EventHandler onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V

minecraft-tfc  |                63: ASM: class com.betterchunkloading.event.EventHandler onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                64: ASM: tfcshipwreck.TfcShipwreckMod@346b1d24 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                65: ASM: tax.taknax.taxdv.TaxdvMod@213c6877 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                66: ASM: tfcoceansvillage.TfcOceansVillageMod@6f8855db tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                67: ASM: tfc_seven_seas.TfcSevenSeasMod@58a576c2 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                68: ASM: tfcorewashing.TfcorewashingMod@1d39135b tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                69: ASM: lithiccoinstrading.LithiccoinstradingMod@6e5e899f tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                70: ASM: tfcitems.TfcItemsMod@11ad20ed tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                71: ASM: tfcspindles.TfcSpindlesMod@201dec53 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                72: ASM: tfcmuskets.TfcMusketsMod@91b6313 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                73: ASM: tfcmetaltools.TfcMetalToolsMod@6e80c93f tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                74: ASM: com.yoda.unstableportals.UnstablePortalsEvents@262f0ae5 onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                75: ASM: tfcmetalitems.TfcMetalItemsMod@3ec54137 tick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                76: ASM: class com.mrh0.createaddition.event.GameEvents serverTickEvent(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                77: ASM: mekanism.common.CommonWorldTickHandler@651ad2ac onTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                78: ASM: Network Registry:
minecraft-tfc  | {} onTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                79: ASM: class com.natamus.manure.forge.events.ForgeManureDropEvent onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                80: ASM: pregenerator.common.manager.ServerManager@d246629 onTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                81: ASM: com.natamus.collective.forge.events.RegisterCollectiveForgeEvents@27f192e4 onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  |                82: ASM: class com.momosoftworks.coldsweat.core.event.TaskScheduler runScheduledTasks(Lnet/minecraftforge/event/TickEvent;)V
minecraft-tfc  |                83: LOWEST
minecraft-tfc  |                84: net.minecraftforge.eventbus.EventBus$$Lambda$1589/0x000071e4fd81c930@249ae601
minecraft-tfc  |                85: ASM: com.github.alexthe666.citadel.ServerProxy@108e698 onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
minecraft-tfc  | java.util.concurrent.CompletionException: net.minecraft.ReportedException: Exception generating new chunk
minecraft-tfc  |        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
minecraft-tfc  |        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
minecraft-tfc  |        at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1159)
minecraft-tfc  |        at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482)
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ChunkTaskPriorityQueueSorter.m_143188_(ChunkTaskPriorityQueueSorter.java:62)

minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.ProcessorMailbox.m_18759_(ProcessorMailbox.java:91)
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.ProcessorMailbox.m_18747_(ProcessorMailbox.java:146)
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.ProcessorMailbox.run(ProcessorMailbox.java:102)
minecraft-tfc  |        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
minecraft-tfc  |        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
minecraft-tfc  |        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
minecraft-tfc  |        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
minecraft-tfc  |        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
minecraft-tfc  |        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
minecraft-tfc  | Caused by: net.minecraft.ReportedException: Exception generating new chunk
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ChunkMap.lambda$scheduleChunkGeneration$27(ChunkMap.java:660)
minecraft-tfc  |        at MC-BOOTSTRAP/datafixerupper@6.0.8/com.mojang.datafixers.util.Either$Left.map(Either.java:38)
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ChunkMap.lambda$scheduleChunkGeneration$29(ChunkMap.java:634)
minecraft-tfc  |        at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150)
minecraft-tfc  |        ... 11 more
minecraft-tfc  | Caused by: java.lang.NullPointerException: Cannot load from int array because "this.surfaceHeight" is null
minecraft-tfc  |        at TRANSFORMER/tfc@3.2.21/net.dries007.tfc.world.chunkdata.RockData.getRock(RockData.java:57)
minecraft-tfc  |        at TRANSFORMER/tfc@3.2.21/net.dries007.tfc.world.chunkdata.RockData.getRock(RockData.java:51)
minecraft-tfc  |        at TRANSFORMER/tfc@3.2.21/net.dries007.tfc.world.feature.LooseRockFeature.m_142674_(LooseRockFeature.java:47)
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.levelgen.feature.Feature.m_225028_(Feature.java:155)
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.levelgen.feature.ConfiguredFeature.m_224953_(ConfiguredFeature.java:28)
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.levelgen.placement.PlacedFeature.m_226362_(PlacedFeature.java:54)
minecraft-tfc  |        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
minecraft-tfc  |        at java.base/java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:411)
minecraft-tfc  |        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
minecraft-tfc  |        at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
minecraft-tfc  |        at java.base/java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:411)
minecraft-tfc  |        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
minecraft-tfc  |        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
minecraft-tfc  |        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
minecraft-tfc  |        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
minecraft-tfc  |        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
minecraft-tfc  |        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.levelgen.placement.PlacedFeature.m_226368_(PlacedFeature.java:53)

minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.levelgen.placement.PlacedFeature.m_226357_(PlacedFeature.java:38)
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.levelgen.feature.RandomPatchFeature.m_142674_(RandomPatchFeature.java:28)
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.levelgen.feature.Feature.m_225028_(Feature.java:155)
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.levelgen.feature.ConfiguredFeature.m_224953_(ConfiguredFeature.java:28)
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.levelgen.placement.PlacedFeature.m_226362_(PlacedFeature.java:54)
minecraft-tfc  |        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
minecraft-tfc  |        at java.base/java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:411)
minecraft-tfc  |        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
minecraft-tfc  |        at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
minecraft-tfc  |        at java.base/java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:411)
minecraft-tfc  |        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
minecraft-tfc  |        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
minecraft-tfc  |        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
minecraft-tfc  |        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
minecraft-tfc  |        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
minecraft-tfc  |        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.levelgen.placement.PlacedFeature.m_226368_(PlacedFeature.java:53)
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.levelgen.placement.PlacedFeature.m_226377_(PlacedFeature.java:42)
minecraft-tfc  |        at TRANSFORMER/tfc@3.2.21/net.dries007.tfc.world.TFCChunkGenerator.m_213609_(TFCChunkGenerator.java:369)
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.chunk.ChunkStatus.m_279978_(ChunkStatus.java:108)
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.chunk.ChunkStatus$SimpleGenerationTask.m_214024_(ChunkStatus.java:309)
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.chunk.ChunkStatus.m_280308_(ChunkStatus.java:252)
minecraft-tfc  |        at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ChunkMap.lambda$scheduleChunkGeneration$27(ChunkMap.java:643)
minecraft-tfc  |        ... 14 more
minecraft-tfc  | 
[09:13:48] [Server thread/ERROR] [minecraft/MinecraftServer]: Encountered an unexpected exception
minecraft-tfc  | java.util.concurrent.CompletionException: net.minecraft.ReportedException: Exception generating new chunk
minecraft-tfc  |        at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315) ~[?:?] {re:mixin}
minecraft-tfc  |        at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320) ~[?:?] {re:mixin}
minecraft-tfc  |        at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1159) ~[?:?] {}
minecraft-tfc  |        at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482) ~[?:?] {}
minecraft-tfc  |        at net.minecraft.server.level.ChunkTaskPriorityQueueSorter.m_143188_(ChunkTaskPriorityQueueSorter.java:62) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,re:classloading,pl:mixin:APP:betterchunkloading.mixins.json:ChunkTaskPriorityQueueSorterMixin,pl:mixin:A}

minecraft-tfc  |        at net.minecraft.util.thread.ProcessorMailbox.m_18759_(ProcessorMailbox.java:91) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading}
minecraft-tfc  |        at net.minecraft.util.thread.ProcessorMailbox.m_18747_(ProcessorMailbox.java:146) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading}
minecraft-tfc  |        at net.minecraft.util.thread.ProcessorMailbox.run(ProcessorMailbox.java:102) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:classloading}
minecraft-tfc  |        at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395) ~[?:?] {}
minecraft-tfc  |        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[?:?] {}
minecraft-tfc  |        at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[?:?] {}
minecraft-tfc  |        at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[?:?] {re:mixin,re:computing_frames}
minecraft-tfc  |        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[?:?] {re:mixin,re:computing_frames}
minecraft-tfc  |        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[?:?] {re:mixin}
minecraft-tfc  | Caused by: net.minecraft.ReportedException: Exception generating new chunk
minecraft-tfc  |        at net.minecraft.server.level.ChunkMap.lambda$scheduleChunkGeneration$27(ChunkMap.java:660) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betterchunkloading.mixins.json:ChunkMapViewDistanceFixedTwo,pl:mixin:APP:modernfix-modernfix.mixins.json:bugfix.paper_chunk_patches.ChunkMapMixin,pl:mixin:APP:modernfix-modernfix.mixins.json:bugfix.chunk_deadlock.ChunkMapLoadMixin,pl:mixin:APP:chunkpregen.mixins.json:common.storage.ChunkManagerMixin,pl:mixin:APP:adastra-common.mixins.json:common.multipart.ChunkMapMixin,pl:mixin:APP:tfc.mixins.json:ChunkMapMixin,pl:mixin:APP:tfc.mixins.json:accessor.ChunkMapAccessor,pl:mixin:APP:railways.mixins.json:ChunkMapAccessor,pl:mixin:APP:ftbchunks-common.mixins.json:ChunkMapMixin,pl:mixin:APP:lychee.mixins.json:ChunkMapAccess,pl:mixin:APP:securitycraft.mixins.json:camera.ChunkMapMixin,pl:mixin:APP:railways-common.mixins.json:conductor_possession.ChunkMapMixin,pl:mixin:A}
minecraft-tfc  |        at com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[datafixerupper-6.0.8.jar%2377!/:?] {}
minecraft-tfc  |        at net.minecraft.server.level.ChunkMap.lambda$scheduleChunkGeneration$29(ChunkMap.java:634) ~[server-1.20.1-20230612.114412-srg.jar%23731!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betterchunkloading.mixins.json:ChunkMapViewDistanceFixedTwo,pl:mixin:APP:modernfix-modernfix.mixins.json:bugfix.paper_chunk_patches.ChunkMapMixin,pl:mixin:APP:modernfix-modernfix.mixins.json:bugfix.chunk_deadlock.ChunkMapLoadMixin,pl:mixin:APP:chunkpregen.mixins.json:common.storage.ChunkManagerMixin,pl:mixin:APP:adastra-common.mixins.json:common.multipart.ChunkMapMixin,pl:mixin:APP:tfc.mixins.json:ChunkMapMixin,pl:mixin:APP:tfc.mixins.json:accessor.ChunkMapAccessor,pl:mixin:APP:railways.mixins.json:ChunkMapAccessor,pl:mixin:APP:ftbchunks-common.mixins.json:ChunkMapMixin,pl:mixin:APP:lychee.mixins.json:ChunkMapAccess,pl:mixin:APP:securitycraft.mixins.json:camera.ChunkMapMixin,pl:mixin:APP:railways-common.mixins.json:conductor_possession.ChunkMapMixin,pl:mixin:A}
minecraft-tfc  |        at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150) ~[?:?] {}
```

I Had manually change getRock method to

```java
    public RockSettings getRock(int x, int y, int z)
    {
        assert generator != null && surfaceHeight != null;
        if(generator != null && surfaceHeight != null){    
            return generator.generateRock(x, y, z, surfaceHeight[Units.index(x, z)], cache);
        } else {
            return generator.generateRock(x, y, z, 200, cache);
        }
    }
```
and it worked for me

--- a/src/main/java/net/dries007/tfc/world/chunkdata/RockData.java
+++ b/src/main/java/net/dries007/tfc/world/chunkdata/RockData.java
@@ -10,6 +10,7 @@ import net.minecraft.core.BlockPos;
 import net.minecraft.world.level.ChunkPos;
 import org.jetbrains.annotations.Nullable;
 
+import net.dries007.tfc.TerraFirmaCraft;
 import net.dries007.tfc.world.region.Units;
 import net.dries007.tfc.world.settings.RockSettings;
 
@@ -53,8 +54,18 @@ public final class RockData
 
     public RockSettings getRock(int x, int y, int z)
     {
-        assert generator != null && surfaceHeight != null;
-        return generator.generateRock(x, y, z, surfaceHeight[Units.index(x, z)], cache);
+        assert generator != null;
+        if (surfaceHeight != null)
+        {
+            return generator.generateRock(x, y, z, surfaceHeight[Units.index(x, z)], cache);
+        }
+        else
+        {
+            // Fallback in case a mod is trying to generate a TFC chunk before the surface height cache is populated,
+            // which has caused many crashes and incompatibilities in the past.
+            TerraFirmaCraft.LOGGER.warn("Queried rock data for position ({}, {}, {}) before surface height cache was populated", x, y, z);
+            return generator.generateRock(x, y, z, 64, cache);
+        }
     }
 
     public int[] getSurfaceHeight()

